Use Ref for m_node in ComposedTreeAncestorIterator

jace0x21 opened the pull request:
#### 15d4f3483d44b391652df7d59e5eba54e9316063
<pre>
Use Ref for m_node in ComposedTreeAncestorIterator
<a href="https://bugs.webkit.org/show_bug.cgi?id=254274">https://bugs.webkit.org/show_bug.cgi?id=254274</a>
rdar://107061402

Reviewed by Chris Dumez.

* Source/WebCore/dom/ComposedTreeAncestorIterator.h:
(WebCore::ComposedTreeAncestorAdapter::ComposedTreeAncestorAdapter):
(WebCore::ComposedTreeAncestorAdapter::begin):

Canonical link: <a href="https://commits.webkit.org/262298@main">https://commits.webkit.org/262298@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/1223faad913e128c55a26836024977f3cd394ec8

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/958 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/985 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/1022 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/1423 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/851 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/947 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/1037 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/1070 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/1059 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/967 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/15/builds/908 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/900 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/1329 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/953 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/897 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/898 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/875 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/924 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/1942 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/916 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/17/builds/991 "Passed tests") | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/863 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/906 "Passed tests") | | 
| [❌ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/287 "Failed to push commit to Webkit repository") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/930 "Built successfully") | | | 
| | | | | 
<!--EWS-Status-Bubble-End-->

cdumez commented on Source/WebCore/dom/ComposedTreeAncestorIterator.h:
```diff
@@ -96,14 +96,14 @@ class ComposedTreeAncestorAdapter {
 
     ComposedTreeAncestorAdapter(Node& node)
-        : m_node(node)
+        : m_node(Ref { node })
```
Not necessary

cdumez commented on Source/WebCore/dom/ComposedTreeAncestorIterator.h:
```diff
@@ -96,14 +96,14 @@ class ComposedTreeAncestorAdapter {
     iterator begin()
     {
-        if (auto shadowRoot = dynamicDowncast<ShadowRoot>(m_node))
+        if (auto shadowRoot = dynamicDowncast<ShadowRoot>(m_node.ptr()))
```
.get() to avoid a null check

cdumez commented on Source/WebCore/dom/ComposedTreeAncestorIterator.h:
```diff
@@ -96,14 +96,14 @@ class ComposedTreeAncestorAdapter {
+        if (auto shadowRoot = dynamicDowncast<ShadowRoot>(m_node.ptr()))
             return iterator(*shadowRoot->host());
-        if (auto pseudoElement = dynamicDowncast<PseudoElement>(m_node))
+        if (auto pseudoElement = dynamicDowncast<PseudoElement>(m_node.ptr()))
```
ditto.

jace0x21 committed:
Use Ref for m_node in ComposedTreeAncestorIterator
https://bugs.webkit.org/show_bug.cgi?id=254274
rdar://107061402

Reviewed by Chris Dumez.

* Source/WebCore/dom/ComposedTreeAncestorIterator.h:
(WebCore::ComposedTreeAncestorAdapter::ComposedTreeAncestorAdapter):
(WebCore::ComposedTreeAncestorAdapter::begin):

Canonical link: https://commits.webkit.org/262298@main
